### PR TITLE
Component header

### PR DESCRIPTION
**Pull request by chae-la**
Two components created: Header and HeaderHome (HeaderHome only used in home page)

### CODE REVIEW

File: src/components/Header/Header.scss [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+       &__title-div{
+        &--title{
+            font-size: 6rem;
+        }
+        &--subtitle{
+            font-size: 11.5rem;
+        }
+       }
+    }
+  }
```
Comment (ample-samples) [inline src/components/Header/Header.scss:125]: The size of the header is far too large on screen sizes from tablet to desktop, ik we don't have wireframes to work from for larger screen sizes so just get this sized sensibly

File: src/components/Header/Header.stories.tsx
Comment (ample-samples) [inline src/components/Header/Header.stories.tsx:1]: There is a good story component for the header, can you add another for the homeHeader please

File: src/components/Header/Header.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import {
+  reactRouterParameters,
+  withRouter,
+} from "storybook-addon-remix-react-router";
+import Header from "./Header";
+
+const meta: Meta<typeof Header> = {
+  title: "Components/Header",
+  component: Header,
+  decorators: [withRouter],
```
Comment (ample-samples) [inline src/components/Header/Header.stories.tsx:11]: This meta variable is missing autodocs, please add `tags: ["autodocs"],` to the meta variable

File: src/components/HeaderHome/HeaderHome.stories.tsx
Comment (ample-samples) [inline src/components/HeaderHome/HeaderHome.stories.tsx:1]: Using npm run storybook, I can't see the HeaderHome component 🤔 can you look into a solution for this please

**Review by ample-samples — CHANGES_REQUESTED** [review]

File: src/components/HeaderHome/HeaderHome.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import "./HeaderHome.scss";
+
+type HeaderHomeProps = {
+  image: string;
+  date?: string;
```
Comment (Charlie-robin) [inline src/components/HeaderHome/HeaderHome.tsx:5]: I think that these should not be optional or if they are optional you need to conditionally render the img and p tag related to them.

File: src/components/HeaderHome/HeaderHome.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+const meta: Meta<typeof HeaderHome> = {
+    title: "Components/HeaderHome",
+    component: HeaderHome,
+  };
+
+  export default meta;
+
+  type Story = StoryObj<typeof HeaderHome>;
+
+export const WithDate: Story = {
```
Comment (Charlie-robin) [inline src/components/HeaderHome/HeaderHome.stories.tsx:13]: It is a simple component so I do not think you need 3 stories for it.

**Review by Charlie-robin — CHANGES_REQUESTED** [review]

File: src/components/HeaderHome/HeaderHome.stories.tsx (hunk 4 above)
Comment (Charlie-robin) [reply src/components/HeaderHome/HeaderHome.stories.tsx:14]: I think you only need this story as that component is only used once and is given all of its props. 

```js
export const WithLocationAndImage: Story = {
    args: {
      date: "29 April 2024",
      location: "Malvern Hills",
      image: "https://s3-alpha-sig.figma.com/img/9003/31aa/6dd32f49cc5bf6f29932b0481a2af856?Expires=1713744000&Key-Pair-Id=APKAQ4GOSFWCVNEHN3O4&Signature=FrjVJj2XFAnoifxYaGoHI0CfbBKprDFMJqDeV1ur21HRwi~j1ZYNRCW-3LPZchYU-brnUydqEOyJ8eUfwA9nB6ATjPomas0~pIPizR9gu2Je79kN9EP1pmgZjI3K6aKUXz7IoMgEErOcHxjXyt42pBlu1Ym8~HfjFPcrLW1igtrvA6JO9y997SJDeldC1ENK0Y8myy9wViW54XoUX1ripSdMlgX9NG2pa--C1P1eafFfIzyQyWrB-Uz8e9eWq0va0JvsDInxsaRT1wOvxHVAxGc~nyK5VJwcSVlplUeTEEL~fMMPrn31K8dbJHlxd0bGEGobas-1vWFzvzv~01QwUw__"
    },
  };
```

This means that you can remove this story as it is not necessary 

```js
export const WithDate: Story = {
  args: {
    date: "29 April 2024",
  },
};
```

File: src/components/Header/Header.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        <Link to="/profile">
+          <img
+            className="header__image-div header__image-div--image"
+            src={profileImage}
+          />
+        </Link>
+      </div>
+
+      <div className="header__title-div">
+        <h4 className="header__title-div header__title-div--title">{title}</h4>
```
Comment (Charlie-robin) [inline src/components/Header/Header.tsx:23]: I don't think this needs to be a prop as it is WAY OF THE VIKING on all of the pages

File: src/components/Header/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+};
+
+const Header = ({ title, subtitle, profileImage }: HeaderProps) => {
+  return (
+    <header className="header">
+      <div className="header__image-div">
+        <Link to="/profile">
+          <img
+            className="header__image-div header__image-div--image"
+            src={profileImage}
```
Comment (Charlie-robin) [inline src/components/Header/Header.tsx:17]: I think that we need to handle if the profileImage is not passed as a prop, perhaps we have a default image to use if it is undefined given?

**Review by Charlie-robin — CHANGES_REQUESTED** [review]

**Review by Charlie-robin — APPROVED** [review]

**Pull request merged**